cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -498,16 +512,22 @@ public void getArtifactPlugin(HttpRequest request, HttpResponder responder,
                                 @PathParam("artifact-version") String artifactVersion,
                                 @PathParam("plugin-type") String pluginType,
                                 @PathParam("plugin-name") String pluginName,
-                                @QueryParam("scope") @DefaultValue("user") String scope)
-    throws NamespaceNotFoundException, BadRequestException, ArtifactNotFoundException {
+                                @QueryParam("scope") @DefaultValue("user") String scope,
+                                @QueryParam("versionRange") String versionRange,
+                                @QueryParam("limit") @DefaultValue("-1") String limit,
+                                @QueryParam("order") String order)
+    throws NamespaceNotFoundException, BadRequestException, ArtifactNotFoundException, InvalidArtifactRangeException {
 
     NamespaceId namespace = Ids.namespace(namespaceId);
     NamespaceId artifactNamespace = validateAndGetScopedNamespace(namespace, scope);
     Id.Artifact artifactId = validateAndGetArtifactId(artifactNamespace, artifactName, artifactVersion);
+    ArtifactRange range = versionRange == null ? null : ArtifactRange.parse(namespace, artifactName, versionRange);
+    int limitNumber = Integer.valueOf(limit);
+    SortInfo.SortOrder sortOrder = order == null ? null : SortInfo.SortOrder.valueOf(order);
 
     try {
       SortedMap<ArtifactDescriptor, PluginClass> plugins =
-        artifactRepository.getPlugins(namespace, artifactId, pluginType, pluginName);
+        artifactRepository.getPlugins(namespace, artifactId, pluginType, pluginName, range, limitNumber, sortOrder);
```

Comment:
may be better for this method to take a Predicate instead of a range, to give you more control over what should be filtered once this endpoint supports filtering by artifact scope, name, and version.